server/lib/transferwise.ts

```diff
@@ -90,55 +90,104 @@ const parseError = (
   return new TransferwiseError(message, code);
 };
 
-export const requestDataAndThrowParsedError = (
-  fn: Function,
+export async function getToken(connectedAccount: ConnectedAccount, refresh = false): Promise<string> {
+  // OAuth token, require us to refresh every 12 hours
+  const tokenCreation = moment.utc(connectedAccount.data.created_at);
+  const diff = moment.duration(moment.utc().diff(tokenCreation)).asSeconds();
+  const isOutdated = diff > <number>connectedAccount.data.expires_in - 60;
+  if (refresh || isOutdated) {
+    const newToken = await getOrRefreshToken({ refreshToken: connectedAccount.refreshToken });
+    if (!newToken) {
+      throw new Error('There was an error refreshing the Transferwise token');
```

Comment:
Would this require reconnecting the wise account? If so, we could send an email on this error

Reply:
That's a very good idea.

Reply:
I'll start by tracking the activity to make sure that it wouldn't spam host admins.